cpp/src/arrow/chunked_array.h

```diff
@@ -141,6 +141,9 @@ class ARROW_EXPORT ChunkedArray {
   /// \brief Return the type of the chunked array
   const std::shared_ptr<DataType>& type() const { return type_; }
 
+  /// \brief Return chunk resolver of the chunked array
+  const internal::ChunkResolver& GetChunkResolver() const { return chunk_resolver_; }
```

Comment:
This is exposing an internal facility as a public member API. I don't know what our API hygiene for this should be. @westonpace @xhochy @lidavidm Thoughts?

(we could also make this private and use a `friend` declaration for the consuming class)

Reply:
At the very least it needs to be clearly marked off as an internal API, right now it looks like any other public API. Ideally it would be private with a friend declaration.

```suggestion
  const internal::ChunkResolver& chunk_resolver() const { return chunk_resolver_; }
```

Reply:
I am in favor of using a `friend` declaration.